### PR TITLE
Integrate CI

### PR DESCRIPTION
**Pull request by vr4b4c**
Task: [#761](https://app.productive.io/1-infinum/projects/134/tasks/task/7565017?filter=MjQxOTMy)

#### Aim
Define CI steps.

#### Solution
- Add RuboCop
- Fix RuboCop offenses
- Define GitHub Actions build workflow
- Remove Travis CI config

### CODE REVIEW

File: shaman.gemspec [hunk 1]
```diff
@@ -13,18 +15,16 @@ Gem::Specification.new do |spec|
   spec.homepage      = 'https://github.com/infinum/shaman'
   spec.license       = 'MIT'
 
+  spec.required_ruby_version = '>= 2.6'
```
Comment (uncoverd) [inline shaman.gemspec]: Do we have a concrete blocker why we can't go lower? I'm not sure what the mobile build pipelines are actually running 🤔

File: shaman.gemspec [hunk 2]
```diff
@@ -13,18 +15,16 @@ Gem::Specification.new do |spec|
...
-  spec.add_development_dependency 'rspec'
-  spec.add_development_dependency 'pry'
-  spec.add_dependency 'http'
-  spec.add_dependency 'git'
   spec.add_dependency 'commander'
+  spec.add_dependency 'git'
+  spec.add_dependency 'http'
   spec.add_dependency 'tty-prompt', '0.12.0'
   spec.add_dependency 'webmock'
+  spec.metadata['rubygems_mfa_required'] = 'true'
```
Comment (uncoverd) [inline shaman.gemspec:28]: 👏

File: shaman.gemspec (hunk 1 above)
Comment (vr4b4c) [reply shaman.gemspec]: I set this based on best judgment and limited analysis (considering only runtime dependencies, which btw don't have version constraints defined, so I assumed versions). Now that I think about it, we should be strict and support only Ruby version > 3. It was released more than 3 years ago, and 2.X reached EOL more than a year ago. Supporting older Ruby versions will also limit our ability to upgrade to newer dependency versions.

Comment (uncoverd) [reply shaman.gemspec]: This would be a breaking change right? In which release would we package this, the upcoming major one?

Comment (vr4b4c) [reply shaman.gemspec]: It's a breaking change and it'll require a major release.

**Pull request merged**